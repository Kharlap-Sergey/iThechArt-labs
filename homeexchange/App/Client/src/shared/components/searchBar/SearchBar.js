import React from "react";
import PropTypes from 'prop-types'
import "./search-bar.scss";

function SearchBar({ onChange, val, onSearchClick, isOpen }) {
  const show = val?.length > 0;
  return (
    <div
      className={`search-bar-container${
        show || isOpen ? " search-bar-container--show" : ""
      }`}
    >
      <input
        type="text"
        placeholder="Search by location..."
        autoFocus={show || isOpen}
        value={val}
        onChange={(e) => onChange(e)}
      />
      <div className="search" onClick={onSearchClick} />
    </div>
  );
}

SearchBar.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  val: PropTypes.string.isRequired,
}
export default SearchBar;
