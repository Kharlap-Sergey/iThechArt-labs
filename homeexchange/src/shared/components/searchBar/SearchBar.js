import React from "react";
import "./search-bar.scss";
function SearchBar() {
  return (
    <div className="search-bar-container">
      <input type="text" placeholder="Search..." />
      <div className="search"></div>
    </div>
  );
}

export default SearchBar;
