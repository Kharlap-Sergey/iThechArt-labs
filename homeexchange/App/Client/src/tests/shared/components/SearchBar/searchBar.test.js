import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "shared/components/searchBar/SearchBar";

test("expected component will have search-bar-container class", () => {
  const component = renderer.create(<SearchBar val="" isOpen={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(component.root.children[0].props.className).toBe(
    "search-bar-container"
  );
});

test("expected component will have open state class", () => {
  const component = renderer.create(<SearchBar val="" isOpen={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(component.root.children[0].props.className).toBe(
    "search-bar-container search-bar-container--show"
  );
});

test("expect component change tree via props", () => {
  let root;
  renderer.act(() => {
    root = renderer.create(<SearchBar val="" isOpen={false} />);
  });

  expect(root.toJSON()).toMatchSnapshot();

  renderer.act(() => {
    root.update(<SearchBar val="" isOpen={true} />);
  });
  expect(root.toJSON()).toMatchSnapshot();
});

test("expect component change val", () => {
  let root;
  renderer.act(() => {
    root = renderer.create(<SearchBar val="" isOpen={true} />);
  });

  expect(root.toJSON()).toMatchSnapshot();

  const someVale = "some lasdjfaklsdf";
  renderer.act(() => {
    root.update(<SearchBar val={someVale} isOpen={true} />);
  });
  expect(root.toJSON()).toMatchSnapshot();
  expect(
    root.root.findByProps({ placeholder: "Search by location..." }).props.value
  ).toBe(someVale);
});
