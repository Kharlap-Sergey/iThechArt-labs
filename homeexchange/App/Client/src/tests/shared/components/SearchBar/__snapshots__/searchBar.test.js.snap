// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect component change tree via props 1`] = `
<div
  className="search-bar-container"
>
  <input
    autoFocus={false}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value=""
  />
  <div
    className="search"
  />
</div>
`;

exports[`expect component change tree via props 2`] = `
<div
  className="search-bar-container search-bar-container--show"
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value=""
  />
  <div
    className="search"
  />
</div>
`;

exports[`expect component change val 1`] = `
<div
  className="search-bar-container search-bar-container--show"
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value=""
  />
  <div
    className="search"
  />
</div>
`;

exports[`expect component change val 2`] = `
<div
  className="search-bar-container search-bar-container--show"
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value="some lasdjfaklsdf"
  />
  <div
    className="search"
  />
</div>
`;

exports[`expected component will have open state class 1`] = `
<div
  className="search-bar-container search-bar-container--show"
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value=""
  />
  <div
    className="search"
  />
</div>
`;

exports[`expected component will have search-bar-container class 1`] = `
<div
  className="search-bar-container"
>
  <input
    autoFocus={false}
    onChange={[Function]}
    placeholder="Search by location..."
    type="text"
    value=""
  />
  <div
    className="search"
  />
</div>
`;
